src/Form/Control/Multiline.php

```diff
@@ -324,24 +344,22 @@ public function saveRows(): self
         $currentIds = array_column($model->export(), $model->id_field);
 
         foreach ($this->rowData as $row) {
-            // should clone model to be able to save it multiple times
-            $row_model = clone $model;
+            $entity = $model->createEntity();
 
             foreach ($row as $fieldName => $value) {
                 if ($fieldName === '__atkml') {
                     continue;
                 }
 
-                if ($fieldName === $row_model->id_field && $value) {
-                    $row_model = $row_model->load($value);
+                if ($fieldName === $model->id_field && $value) {
+                    $entity = $model->load($value);
                 }
 
-                $field = $row_model->getField($fieldName);
-                if ($field->isEditable()) {
-                    $field->set($value);
+                if ($model->getField($fieldName)->isEditable()) {
+                    $entity->set($fieldName, $value);
                 }
             }
-            $id = $row_model->save()->getId();
+            $id = $entity->save()->getId();
```

Comment:
save is outside foreach, is that intended?

otherwise feel free to merge it

Reply:
Yes - the intention is to collect all record values for one row into the for loop and save after.

Reply:
but a few lines above there is a `load`, thus I think "collect all record values for one row" can not work, at least the changes will be saved for one row only

Reply:
bad explanation from me. 

The data collect from load contains all rows and each row has one set of record. 
The second for loop is too loop for each field and check if the current field is editable prior to set is value. 
Once all fields are set within the row, then the record is save.

Hope it is clearer now... 

Maybe we could use $entity->save($row) directly but I am not sure how Data will react if trying to pass a field => value set to a non editable field... This is were my knowledge of Data fails 😄 Multiline will display the non editable field as readonly input.

Reply:
AFAK this relies on :N typecasting from single cell value which... Which at least is not good to be saved n times when looping.

feel free to merge without my review once you are happy with the changes ;-)